### ISSUE TITLE
bot does not capture "/me" (status) IRC messages in logs

### ISSUE THREAD

**Issue by brainwane** [pmxbot/pmxbot#75]
What just happened in `#pypa-dev` on Freenode:

~~~
<jaraco> !thanks sumanah for letting me know.
<pmxbot> you're doing good work, sumanah!
* sumanah basks
<sumanah> looking at https://wiki.python.org/psf/PackagingWG/2018-03-06-Warehouse before today's meeting
~~~

What shows up in the logs at http://kafka.dcpython.org/day/pypa-dev/2018-03-12 :

~~~
[14:49:52] <jaraco> !thanks sumanah for letting me know.
[14:49:52] <pmxbot> you're doing good work, sumanah!
[14:53:25] <sumanah> looking at wiki.python.org/psf/PackagingWG/2018-03-06-Warehouse before today's meeting 
~~~

**Comment by jaraco** [comment]
Indeed, that's true, and that's been the case since the inception, so the behavior is (incidentally) by design. I do agree it's somewhat surprising behavior. I'm not sure if the behavior was omitted intentionally or unintentionally or if there would be issues with adding support for it.

I believe the fix would be within pmxbot.irc.LoggingCommandBot to add the appropriate `on_action` and dispatch the message the same as one would `on_privmsg` (maybe prepending `/me ` to it).

Is this something you'd be interested in contributing to?

**Comment by brainwane** [comment]
I'm sorry -- I wish I could contribute and fix this, but I need to say no-for-now and concentrate on Twine. Thanks for the suggestion, though!

**Comment by brainwane** [comment]
Thanks!

**Comment by brainwane** [comment]
@jaraco at least based on today's conversation in http://kafka.dcpython.org/day/pypa-dev/2020-04-13 I think this behavior is not entirely fixed. The actual conversation went:

```
<sumanah> Hi pradyunsg - hope you are doing all right
<pradyunsg> sumanah: I am, and you?
<sumanah> pradyunsg: yup!
* pradyunsg wonders if poetry maintainers have ever thought about the project joining PyPA.
<sangy> pradyunsg: poetry?
<sumanah> pradyunsg: I'd like to know whether Sebastien has considered it, too.
```

but [the log](http://kafka.dcpython.org/day/pypa-dev/2020-04-13) does not include the line in the middle.

**Comment by jaraco** [comment]
I doubt we'll ever get around to root-causing this issue. In 0656b85, I've added a test showing that the bot does log a message if it shows up as a PRIVMSG, but it's not clear to me if or how the bot will ever receive messages that aren't of that type. I'd previously thought they would be handled with the `on_action` handler, but that seems not to be the case. And unfortunately, the IRC protocol is inscrutable and now in many ways obviated by other systems.

Incidentally, while looking into this again, I noticed that [other systems intentionally don't log such activity](https://www.w3.org/wiki/IRC#Out_of_band_or_off_the_record_comments).

Given my lack of interaction with IRC and pmxbot, I'm going to have to close this as won't fix. That said, I welcome other interested parties to revive the conversation and investigate deeper and develop a robust solution.
